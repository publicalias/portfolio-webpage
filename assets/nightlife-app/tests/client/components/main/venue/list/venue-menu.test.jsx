"use strict";

//local imports

const VenueMenu = require("../../../../../../scripts/client/components/main/venue/list/venue-menu");

const { testWrapper } = require("../../../../test-helpers");

//global imports

const { initTestSnapshot } = require("redux/tests/client-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(VenueMenu);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(VenueMenu));

//venue menu

describe("VenueMenu", () => {

  const testSnapshot = initTestSnapshot(testShallow);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (search)", () => testSnapshot({ venues: { list: { menu: { search: "Coffee" } } } }));

});
