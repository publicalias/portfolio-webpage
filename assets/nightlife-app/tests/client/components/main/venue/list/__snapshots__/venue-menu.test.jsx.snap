// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`venue menu should match snapshot 1`] = `
<div
  className="c-list-menu"
>
  <SearchBar
    local={
      Object {
        "handleChange": [Function],
        "handleClear": [Function],
        "handleSubmit": [Function],
        "value": "",
      }
    }
  />
  <MetaDropdown
    local={
      Object {
        "bool": false,
        "handleSelect": [Function],
        "handleToggle": [MockFunction],
        "list": Array [
          Array [
            "05",
            "range",
            "05",
          ],
          Array [
            "10",
            "range",
            "10",
          ],
          Array [
            "15",
            "range",
            "15",
          ],
          Array [
            "20",
            "range",
            "20",
          ],
          Array [
            "25",
            "range",
            "25",
          ],
        ],
        "name": "Range",
      }
    }
  />
  <MetaDropdown
    local={
      Object {
        "bool": false,
        "handleSelect": [Function],
        "handleToggle": [MockFunction],
        "list": Array [
          Array [
            "Match",
            "sort",
            "best_match",
          ],
          Array [
            "Distance",
            "sort",
            "distance",
          ],
          Array [
            "Rating",
            "sort",
            "rating",
          ],
        ],
        "name": "Sort By",
      }
    }
  />
</div>
`;
